### PR TITLE
Enable docker authentication for private images

### LINKED ISSUE #70: docker auth

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix #70 

Added a new config in secrets.yml which looks like this
```yml
docker:
    username: <username>
    password: <password>
    endpoint: <alternative_endpoint> # If not docker hub
```

### CODE REVIEW

**Comment by timotheeguerin** [comment]
Updated the docs to talk about this

File: docs/12-docker-image.md [hunk 1]
```diff
@@ -29,12 +29,28 @@ azb spark cluster create ... --docker-repo jiata/thunderbolt:0.1.0-spark1.6.3-py
...
 
 Please refer to ['../docker-image'](../docker-image) for more information on building your own image.
 
 Once you have your Docker image built and hosted publicly, you can then use the **--docker-repo** parameter in your **azb spark cluster create** command to point to it.
 
 ## Using a custom Docker Image that is Privately Hosted
 
-_Coming soon_
+To use a private docker image you will need to provide a docker username and password that have access to the repository you want to use.
+We also support custom docker endpoint(Not dockerhub)
```
Comment (paselem) [inline docs/12-docker-image.md]: "(Not dockerhub)"

Do we explicitly not support dockerhub at the moment?

File: docs/12-docker-image.md [hunk 2]
```diff
@@ -29,12 +29,28 @@ azb spark cluster create ... --docker-repo jiata/thunderbolt:0.1.0-spark1.6.3-py
...
+
+In `.thunderbolt/secrets.yaml` setup your docker config
+```yaml
+docker:
+    username: <myusername>
+    password: <mypassword>
+```
+
+If your private repository is not on docker hub(Azure container registry for example) you can provide the endpoint here too
```
Comment (paselem) [inline docs/12-docker-image.md]: missing space after 'hub'

File: dtde/clusterlib.py [hunk 3]
```diff
@@ -126,7 +127,7 @@ def generate_cluster_start_task(
             name="SPARK_JUPYTER_PORT", value=spark_jupyter_port),
         batch_models.EnvironmentSetting(
             name="SPARK_WEB_UI_PORT", value=spark_web_ui_port),
-    ]
+    ] + get_docker_credentials()
```
Comment (paselem) [inline dtde/clusterlib.py]: ? Add this on a new line?

File: dtde/clusterlib.py [hunk 4]
```diff
@@ -214,6 +232,11 @@ def create_cluster(
 
     # Wait for the app to finish
     if wait:
+        # util.wait_for_all_nodes_state(pool, frozenset(
```
Comment (paselem) [inline dtde/clusterlib.py]: remove

File: dtde/clusterlib.py [hunk 5]
```diff
@@ -287,9 +310,9 @@ def pretty_node_count(cluster: Cluster) -> str:
 
 
 def pretty_dedicated_node_count(cluster: Cluster)-> str:
-    if (cluster.pool.allocation_state is batch_models.AllocationState.resizing\
-        or cluster.pool.state is batch_models.PoolState.deleting)\
-        and cluster.dedicated_nodes != cluster.target_dedicated_nodes:
+    if (cluster.pool.allocation_state is batch_models.AllocationState.resizing
+            or cluster.pool.state is batch_models.PoolState.deleting)\
```
Comment (paselem) [inline dtde/clusterlib.py]: you removed the \ on line 313, but not on 314. Any reason in particular?

File: dtde/clusterlib.py (hunk 3 above)
Comment (timotheeguerin) [reply dtde/clusterlib.py]: you'd have to have a \ at the end of the line which would look ugly here

File: docs/12-docker-image.md (hunk 1 above)
Comment (timotheeguerin) [reply docs/12-docker-image.md]: Maybe this is a bit confusing. Meant you we also support a custom enpoint that is not dockerhub but its docker hub by default

**Pull request merged**